stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt

```diff
@@ -192,22 +196,23 @@ public class ChannelListViewModel(
         /**
          * We clean up any previous loads to make sure the current one is the only one running.
          */
-        if (queryJob != null) {
-            queryJob?.cancel()
+        queryJob?.cancel()
+        val queryJob = SupervisorJob(viewModelScope.coroutineContext.job).also {
```

Comment:
Why do we specifically create a supervisor here with a parent job context? A supervisor job should override the parent job anyway so it doesn't do anything. We can still pass in the context of the ViewModelScope.

Alternatively, the `ViewModelScope`'s context should already be using a SupervisorJob, so we scan just use that (probably).

Either way - unless we use the job to cancel the children at some point in time, we don't need a supervisor.

If we use it just for that, then we can always have a live SupervisorJob, and instead of canceling the job, we can cancel its children and reuse the job.

Reply:
On the one hand, we use `viewModelScope.coroutineContext.job` as parent for this `SupervisorJob` to ensure this `SupervisorJob ` will be cancelled with `viewModelScope`.

On the other hand, filtering has a shorter lifecycle, because it can be changed with a new `setFilters` method. So this `queryJob` is also passed to all `asLiveData()` methods as well to cancel those. This is because we kept listening to the previous `QueryChannelsState` changes.
For that purpose, a new extension function is being created named `addFlow`.

Reply:
I firstly wanted to reuse a single instance of `QueryChannelsState`. However, in this case,` invokeOnCompletion` won't be called in the snippet below:
```kotlin
internal fun <T, S> MediatorLiveData<T>.addFlow(
    context: CoroutineContext = EmptyCoroutineContext,
    source: Flow<S>,
    onChanged: Observer<in S>
) {
    source.asLiveData(context).also { liveData ->
        addSource(liveData, onChanged)
        context[Job]?.invokeOnCompletion {
            removeSource(liveData)
        }
    }
}
```

Reply:
On the first point - good point, I missed how the parent job is being used (I'm dumb).

On the second - if we have a supervisor job and we pass it around to create new coroutines (children) then we can just cancelChildren right?

If we don't need that, we don't need a supervisor right, we can just pass in a regular job whose parent context is the VM context - then by cancelling the job, we clear up its coroutines and the parent job is not canceled since it's also a supervisor.

So the proposed strucutre would be:

Supervisor Job (VM Scope)
  - Job (query)
      - Any other coroutines started with it
      
Rather than

SupervisorJob (VM Scope)
  - Supervisor (Query)
    - Other coroutines

Does this make sense? I might be missing something but I think we can just rely on a simple job here or use the supervisor in a reusable manner rather than recreating it

Reply:
Makes sense. 
Thanks for the feedback, man 🍻

Replaced `SupervisorJob` with a regular `Job`.